node/runtime/src/lib.rs

```diff
@@ -188,7 +188,7 @@ impl transaction_payment::Trait for Runtime {
 	type OnTransactionPayment = DealWithFees;
 	type TransactionBaseFee = TransactionBaseFee;
 	type TransactionByteFee = TransactionByteFee;
-	type WeightToFee = WeightToFee;
+	type WeightToFee = LinearWeight<TransactionWeightFee>;
```

Comment:
I see you like parameter types, and I'm ok now. But we can keep in mind that this could be written
`LinearWeightToFee::new(1_000)`  or `LinearWeightToFee { coefficient: 1_000 }`

Reply:
I also tried this approach but couldn't get it to compile. @kianenigma [replied above](https://github.com/paritytech/substrate/pull/3856#issuecomment-544254122)

> I don't think you should pursue this. Using self in Convert will not work as it is called as a type function like T::AssociatedType::function(), hence will never have self.

So I'll stick with the type parameter for now. Would be happy to learn if this approach can work though, because I do like it better.

Reply:
no indeed I forget it was just a type not a variable